### ISSUE TITLE
Expose an unlimited refresh token to consumer call of MSAL

### ISSUE THREAD

**Issue by Deymoun** [AzureAD/microsoft-authentication-library-for-js#3122]
## Library
- [x] `@azure/msal-browser@2.11.0`

## Description

Hi,

I already talked about this issue with Juan Balmori Labra from MS Outlook team and Matt Geimer from MS SSO team and they suggested to post here.

I am dealing with the authentication for an Office JS add-in in Outlook.
The [Office JS API](https://docs.microsoft.com/en-us/javascript/api/office-runtime/officeruntime.auth?view=excel-js-preview#getAccessToken_options_) provides a function to get a SSO assertion token for Microsoft Graph API, then the [On-Behalf-Of flow](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-on-behalf-of-flow#saml-assertions-obtained-with-an-oauth20-obo-flow) provides a nested Graph token which contains a refresh token inside.
This [refresh token lifetime](https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-configurable-token-lifetimes#configurable-token-lifetime-properties-after-the-retirement) is also unlimited, which is a key feature since it will be used by the server hosted by my company (out of Outlook process) to manipulate calendar appointments for its clients.
 
My problem is that I read that the [AppSource certification process](https://docs.microsoft.com/en-us/office/dev/store/submit-to-appsource-via-partner-center#certification-process) of the Office JS add-in requires a non-SSO fallback authentication.
So I thought about using msal-browser version 2 which implements the Authorization Code flow in a SPA, however there are two issues:
1.	the refresh token is not exposed,
2.	the refresh token has a [limited lifetime of 24 hours](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/token-lifetimes.md#refresh-tokens).

I read in [another issue](https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/3019#issuecomment-778369931) that the refresh token was exposed in the previous version of authentication library ADAL. 
Would it be possible that MSAL exposes also the refresh token to consumer call, and that it never expires?

Thank you,
Julien.

## Source
- [ ] Customer request

**Comment by jasonnutter** [comment]
@Deymoun The refresh tokens in MSAL.js v2 are short-lived as a security consideration for browsers, and MSALs by design do not return refresh tokens to applications. Can you provide more details about why you need the refresh token?

You can see a sample Office Addin using MSAL.js v2 here: https://github.com/OfficeDev/PnP-OfficeAddins/tree/master/Samples/auth/Office-Add-in-Microsoft-Graph-React

**Comment by Deymoun** [comment]
Hi @jasonnutter my company would need the refresh token to access its clients' calendar anytime after a single user interaction.
The goal is to not have the clients reopen the Office taskpane every day to send again the Graph token, but rather they open **only once** the Office taskpane and send only once the refresh token so that my company can take care of the calendars in a job, **months or years** after this original setup.

**Comment by jasonnutter** [comment]
@Deymoun With MSAL.js, if your app needs an access token, it should invoke `acquireTokenSilent` and MSAL will handle getting a new token using the refresh token (or from the cache). As noted, these refresh tokens are only good for 24 hours. Furthermore, refresh tokens acquired via MSAL.js v2 must be used client-side and cannot be used server-side. Given we do not have plans to remove these restrictions, it sounds like MSAL.js may not be the right library for your scenario, and you may instead want to look at MSAL.net, MSAL Node, etc. depending on your backend auth stack.

**Comment by Deymoun** [comment]
Hi @jasonnutter thanks for your suggestions, however both MSAL.NET and MSAL Node do not expose refresh token ([proof for MSAL Node](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1234#issuecomment-506442939))
Would there be another solution than implementing [the complex solution developed by another consumer of MSAL](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1234#issuecomment-506442939) ?

**Comment by tnorling** [comment]
@Deymoun The decision was made by all MSALs, not just javascript, to not expose the refresh token. As @jasonnutter described, when your application needs an access token it should call `acquireTokenSilent` which will fetch tokens from cache or attempt to renew using the refresh token. If you use a server side library such as msal.net or msal-node the refresh token lifetime will be much longer than 24 hours (weeks or even months) so your users will not need to reauthenticate frequently.

Closing as this feature will not be picked up but feel free to open a new issue if you have further questions. Thanks!

**Comment by tabeth** [comment]
@tnorling @jasonnutter 

Hi to the both of you. I'm pretty sure I know the answer to this [per the documentation](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/token-lifetimes.md), but I wanted to get clarification: 

MSALJS 2.0 (for the browser) doesn't respect the refresh token settings as specified in the User Flow Policy in Azure AD B2C, correct? Meaning that if you have your refresh token set to expire in 90 days, the library will regardless expire your session and require interaction after 48 hours (I believe there's a 24 hour grace period in which you could return to a site using this library and it would renew your session) right?
